rocky/reports/views/generate_report.py

```diff
@@ -146,22 +146,35 @@ def get(self, request, *args, **kwargs):
 
     def generate_reports_for_oois(self) -> Dict[str, Dict[str, Dict[str, str]]]:
         report_data = {}
+        error_oois = []
         for ooi in self.selected_oois:
             report_data[ooi] = {}
-            for report_type in self.get_report_types_from_choice():
-                if Reference.from_str(ooi).class_type in report_type.input_ooi_types:
-                    report = report_type(self.octopoes_api_connector)
-                    data = report.generate_data(ooi, valid_time=self.valid_time)
-                    template = report.template_path
-                    report_data[ooi][report_type.name] = {"data": data, "template": template}
+            try:
+                for report_type in self.get_report_types_from_choice():
+                    if Reference.from_str(ooi).class_type in report_type.input_ooi_types:
+                        report = report_type(self.octopoes_api_connector)
+                        data = report.generate_data(ooi, valid_time=self.valid_time)
+                        template = report.template_path
+                        report_data[ooi][report_type.name] = {"data": data, "template": template}
+            except Exception:
+                error_oois.append(ooi)
+        # If OOI could not be found or the date is incorrect, it will be shown to the user as a message error
+        if error_oois:
+            oois = ", ".join(list(set(error_oois)))
```

Comment:
The extra `list` call is unnecessary, `join` accepts iterables and `set` is an iterable